### ISSUE TITLE
❓ 🙋  NM can in fact de-aggro

### ISSUE THREAD

**Issue by TeoTwawki** [DerpyProjectGroup/topaz#5462]
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## OS / platform the server is running (if known)
n/a

## Branch affected by issue 

`base`

## Steps to reproduce

[](https://github.com/LandSandBoat/server/blob/f0d23cacf13df0a8cc91f8a5342b3351b26fd86a/src/map/entities/mobentity.cpp#L359)

So as you can see [here](https://github.com/LandSandBoat/server/blob/f0d23cacf13df0a8cc91f8a5342b3351b26fd86a/src/map/entities/mobentity.cpp#L359), no NM on LSB can ever be "lost" by the player. 

## Expected behavior

While it can be quite difficult to lose most NM on retail you can in fact lose them. Or at least some of them. I've Lost worm NMs. They forgot I existed and went back underground when I backed out of casting range to rest MP. Likewise kiting a few, I've had them give up on me and start walking home after an extended period of neither of us hitting each other (5 full min) and my being well out of its range (flee).

At the very least, its not a universal behavior for perma-aggro. In BCNM it is, but not out in outdoor zones.

**Comment by sarca571ca** [comment]
Just a few things off the top of my head.
- When you lost them did you cross over water? <- Maybe in 5 minutes also they lose your scent?
- Is it isolated to certain families? <- could some families chase you longer?
- Potentially themed? Like a beastmen nm protecting an area?

**Comment by TeoTwawki** [comment]
>     * When you lost them did you cross over water? <- Maybe in 5 minutes also they lose your scent?

nope. my specific case for opening the issue was a worm, I stayed out of range to long to recovered and he noped out of our fight and went back to idle.

I've also "lost" some NM while kiting before. Specifically void**walker** antlion (not to be confused with void**watch**). Managed to reclaim him but his AI just stopped following me and went idle, I was worried he was gonna go back underground and full heal.

>     * Is it isolated to certain families? <- could some families chase you longer?

very possible

>     * Potentially themed? Like a beastmen nm protecting an area?

might be the case for some. I've lost that one antican ranger NM in altepa when I had to make a run for it after eagle eye shot left me with nearly no HP. The intruder earring NM.



I should point out its a very small number of cases I've witnessed an NM forgetting on retail. I think its similar to the despawn period - some NMs if you wipe there's like a 5 min timer in which your group can reclaim and try again. I think this is like that, but its a very long stretch of neither you nor the NM being able to perform any actions on each other. Each time it happened, I was losing and had taken on something I wasn't strong enough to normally win. I like to sometime try things at lower levels than anyone should.

**Comment by sarca571ca** [comment]
I guess maybe we should maybe implement a flag or something we can set on nm's that have this behavior and sort them out as we go. 

But we'd have to determine why they do it. I suppose some mobs do the same thing though. Running around CN yesterday and noticed some mobs stopped chasing me after a bit.

**Comment by TeoTwawki** [comment]
normal mobs give up for pretty much the same reasons NMs would. I dunno fi it takes as long or if it takes longer for NMs in general for it to happen on, or how many might truly be impossible to "lose". for some it'd be easy to test, for others very hard.

Its just rn in LSB's code there is a specific check that says if the mob is an NM, never deaggro. and since I've seen it happen, I knew that tot be not a universally true thing and opened the issue. Likely instead of asking if the mob is an NM we should have a mobmod and check the state of that mobmod. I failed to link the code where I intended to when I initially posted, had to exit to add the link.

We could default it to the same behavior we have now for NMs. When mobs are set up in core that have a few "default mobmods" that set up things like casting cooldowns for mages so it'd be easy to automatically include one where the ability to deaggro is pre-enabled for normal and pre-disabled for NMs, and swapping that is flipping a switch: change value of the mobmod. This would provide the least amount of "shake up" for mob behavior while we look for further data.
